### ISSUE TITLE
Database column with complex json can cause a segfault

### ISSUE THREAD

**Issue by jdoconnor** [oven-sh/bun#3991]
### What version of Bun is running?

0.7.2

### What platform is your computer?

Darwin 22.5.0 arm64 arm

### What steps can reproduce the bug?

Run the following

```typescript

import { Database } from "bun:sqlite";
const db = new Database(":memory:");
db.prepare(
    `CREATE TABLE IF NOT EXISTS users (
    id TEXT PRIMARY KEY,
    xx TEXT)
`).run();

db.prepare(
    `insert into users (id, xx) values (
    'foobar',
    '{
        "links": [{"1": {
    "2": "https://foobar.to/123",
    "3": "4"
    }}]

    }'
)`).run();

let x = await db.query(
    `SELECT * FROM users
        WHERE users.id = 'foobar'
        limit 1`).get();
console.log(x)
console.log(x.xx)
```

### What is the expected behavior?

the text from column `xx` should be printable. 

### What do you see instead?

segfault

### Additional information

simple json in the column appears to work. Complex json with the link causes this. I have not been able to repro this on 0.7.1.

**Comment by Jarred-Sumner** [comment]
This is probably the same as https://github.com/oven-sh/bun/issues/3977, but I really appreciate the reproduction because that makes this a lot easier to fix.

**Comment by Jarred-Sumner** [comment]
I see the cause and it is indeed a regression in v0.7.2. Will have a fix shortly.

**Comment by Jarred-Sumner** [comment]
Fixed in 637a38f394704eaea29b503a69d554b5726d6214. Please upgrade via `bun upgrade --canary` in an hour or two to get the update before it's released.
